Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/SwapShiftController.cs

```diff
@@ -191,14 +199,10 @@ public static string GenerateNewGuid()
                         SwapShiftObj swapShiftObj = new SwapShiftObj
                         {
                             // Convert shift time to Kronos time zone.
-                            Emp1FromDateTime = this.utility.UTCToKronosTimeZone(
-                                senderShiftDetails.SharedShift.StartDateTime),
-                            Emp1ToDateTime = this.utility.UTCToKronosTimeZone(
-                                senderShiftDetails.SharedShift.EndDateTime),
-                            Emp2FromDateTime = this.utility.UTCToKronosTimeZone(
-                                recipientShiftDetails.SharedShift.StartDateTime),
-                            Emp2ToDateTime = this.utility.UTCToKronosTimeZone(
-                                recipientShiftDetails.SharedShift.EndDateTime),
+                            Emp1FromDateTime = this.utility.UTCToKronosTimeZone(senderShiftDetails.SharedShift.StartDateTime, mappedTeam.KronosTimeZone),
```

Comment:
do you need to check if mappedTeam is null first?

Reply:
Modified to use the same approach as the OpenShiftRequestController to fall back to the global app setting for the time zone, however, as mentioned the mappings are fundamental to the working of the connector and if they are missing the connector will probably not work at all for the team.